features/accommodations/creation/LocationForm.tsx

```diff
@@ -0,0 +1,72 @@
+import { ChangeEvent, type FC } from "react";
+import { SelectOptionField } from "../../../core/components/SelectOptionField";
+import { countries } from "../useCountries";
+import { type CreateAccommodation } from "../createAccommodation";
+import { type Location } from "../AccommodationModels";
+
+type LocationFormProps = {
+  updateFields: (fields: Partial<CreateAccommodation>) => void;
+  location: Location;
+};
+
+export const LocationForm: FC<LocationFormProps> = ({ updateFields, location }) => {
+  const updateLocation = (fields: Partial<Location>) =>
+    updateFields({ location: { ...location, ...fields, saved: false, latitude: 0, longitude: 0 } });
+
+  return (
+    <>
+      <SelectOptionField
+        label="Country"
+        placeholder="Select one"
+        options={countries.map((c) => ({
+          value: c,
+          label: c,
+        }))}
+        formElement={{
+          onChange: (e: ChangeEvent<HTMLInputElement>) =>
+            updateLocation({ country: e.target.value }),
+          required: true,
+        }}
+        value={location.country}
+      />
+      <label className="block text-lg font-medium text-gray-700 mt-5">State</label>
+      <input
```

Comment:
Could we extract this label + input into some component? I don't like that we will have two different types of inputs but it is hard to make a single component for all use cases.

Reply:
This is the same style as "InputField", but without the formElement and errorMessage, as I don't use validation here. But sure, I will extract it.